torch/csrc/jit/passes/to_batch.cpp

```diff
@@ -1,63 +1,441 @@
 #include "torch/csrc/jit/passes/to_batch.h"
 #include "torch/csrc/jit/script/compiler.h"
+#include "torch/csrc/jit/tensor_conversions.h"
 
 namespace torch { namespace jit {
 
 std::unordered_map<std::string, std::shared_ptr<Graph>> ToBatch::batch_operator_table;
 
-void ToBatch::toBatch(Block* block, Block* res_block) {
-  // change inputs of a graph - expand tensor to {data, mask, dims}
+// replace aten operator node with BatchTensor operator graph
+void ToBatch::visitAten(Node* n, Block* block, Block* res_block, std::unordered_map<std::string, Value*>& var_map){
+  if(n->outputs().size() > 1){
+    throw std::runtime_error("NYI: multiple assignment is not supported yet");
+  }
+  auto batch_graph = batch_operator_table.at(n->kind().toUnqualString());
+  std::vector<Value*> new_inputs;
+  for(Value *input : n->inputs()){
+    if(batch_map.find(input) != batch_map.end()){
+      auto new_input = batch_map.at(input);
+      new_inputs.insert(new_inputs.end(), new_input.begin(), new_input.end());
+    }
+    else{
+      throw std::runtime_error("NYI: non-tensor input for aten operator is not supported yet");
+    }
+  }
+  auto outputs = script::inlineCallTo(*res_block->owningGraph(), *batch_graph, new_inputs);
+
+  // do update on assignment
```

Comment:
The graph is in single-static assignment form. Because of this there are no assignments anywhere in the graph. I there is some confusion here because of it. Just because a uniqueName shares a prefix does not mean it is the same variable, uniqueNames are for debugging only. In fact, you can treat every `Value*` as a separate variable, and we should not need to call update.

This will also allow you to remove getNameBaseSuffix usages.